bittide/src/Clash/Explicit/Reset/Extra.hs

```diff
@@ -0,0 +1,48 @@
+-- SPDX-FileCopyrightText: 2022-2023 Google LLC
+--
+-- SPDX-License-Identifier: Apache-2.0
+
+module Clash.Explicit.Reset.Extra where
+
+import Clash.Explicit.Prelude
+import Clash.Cores.Xilinx.Xpm.Cdc.Single
+
+-- | Configuration value to indicate whether resets should be asserted or
+-- deasserted. Used throughout this module.
+data Asserted = Asserted | Deasserted
+
+-- | A reset synchronizer based on 'xpmCdcSingle'. I.e., a reset synchronizer that
+-- is recognized by Vivado as a safe CDC construct.
+xpmResetSynchronizer ::
+  (KnownDomain src, KnownDomain dst) =>
```

Comment:
The `Reset` of the source domain _must_ be synchronous. Since you already bumped Clash for the new reset related functions, you can use:
```suggestion
  (HasSynchronousReset src, KnownDomain dst) =>
```